Plutarch/Lift.hs

```diff
@@ -89,9 +90,15 @@ plift' prog = case evalScript (compile prog) of
 plift :: forall p. (HasCallStack, PLift p) => ClosedTerm p -> PLifted p
 plift prog = case plift' prog of
   Right x -> x
-  Left LiftError_FromRepr -> error "plift failed because of pconstantFromRepr"
-  Left (LiftError_ReadKnownError _) -> error "plift failed because of an internal error in plutus-core"
-  Left (LiftError_EvalError e) -> error $ "plift failed because of an erring term: " <> show e
+  Left LiftError_FromRepr -> error "plift failed: pconstantFromRepr returned 'Nothing'"
+  Left (LiftError_ReadKnownError (ErrorWithCause e causeMaybe)) ->
+    let unliftErrMaybe = e ^? _UnliftingErrorE
+     in error $
+          "plift failed: internal plutus-core error: "
+            <> maybe "ReadKnownEvaluationFailure" show unliftErrMaybe
```

Comment:
bit of an assumption here, `ReadKnownError` (the type of `e`) has 2 unexported constructors. One of them corresponds to `AsUnliftingError` (i.e `_UnliftingErrorE` prism) and the other corresponds to `AsEvaluationFailure` (not to be confused with `AsEvaluationError`). The latter is only useful to *confirm* that the constructor is indeed `ReadKnownEvaluationFailure` - as it holds no fields, and no real information.

Thus, if `unliftErrMaybe` is `Nothing`, the only possibility for `e` is to be `ReadKnownEvaluationFailure` - which is what I'm defaulting to here, without ever invoking `_EvaluationFailure :: Prism' ReadKnownError ()`.

With the lack of pattern matching, exhaustion checks are no longer simple and if they add another constructor - this kinda flies under our nose, potentially. Is there a way to achieve such exhaustion safety with prisms?

Reply:
`ReadKnownEvaluationFailure` should never happen FWIW because we do `evalScript` AFAICT

Reply:
Interesting, should the `Nothing` case use a different string than `ReadKnownEvaluationFailure` perhaps (to indicate that case should be absurd)?

Still doesn't solve the constructor exhaustion issue, but I don't think we can actually solve that. Might just have to put a note here.

Reply:
Rather than `"internal plutus-core error"`, it should be something like `"incorrect type"`.

Reply:
Sure, changed.